### PR TITLE
Create and fetch conversation topics

### PR DESCRIPTION
**Pull request by donatKapesa**

### CODE REVIEW

File: src/main/java/com/google/codeu/data/Message.java [hunk 1]
```diff
@@ -26,25 +26,31 @@
...
+    this(UUID.randomUUID(), user, text, System.currentTimeMillis(), imageUrl, conversationTopicId);
+  }
+
+  public Message(UUID id, String user, String text, long timestamp, String imageUrl, String conversationTopicId) {
     this.id = id;
     this.user = user;
     this.text = text;
     this.timestamp = timestamp;
     this.imageUrl = imageUrl;
+    this.conversationTopicId = conversationTopicId;
```
Comment (bluedragon2) [inline src/main/java/com/google/codeu/data/Message.java:53]: Not all messages will have conversationTopicId, right?
You may want to create an overloaded constructor.
Ideally, you could use the builder pattern: good tutorial here https://dzone.com/articles/design-patterns-the-builder-pattern

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 2]
```diff
@@ -44,12 +44,24 @@ public void storeMessage(Message message) {
 		Entity messageEntity = new Entity("Message", message.getId().toString());
 		messageEntity.setProperty("user", message.getUser());
 		messageEntity.setProperty("text", message.getText());
-		messageEntity.setProperty("timestamp", message.getTimestamp());
+		messageEntity.setProperty("timestamp", message.gettimestamp());
```
Comment (bluedragon2) [inline src/main/java/com/google/codeu/data/Datastore.java:47]: getTimestamp should be the correct convention

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 3]
```diff
@@ -44,12 +44,24 @@ public void storeMessage(Message message) {
...
 		messageEntity.setProperty("imageUrl", message.getImageUrl());
+		messageEntity.setProperty("conversationTopic", message.getConversationTopicId());
```
Comment (bluedragon2) [inline src/main/java/com/google/codeu/data/Datastore.java:49]: ConversationTopicId and imageUrl could be null, right ?

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 2 above)
Comment (donatKapesa) [reply src/main/java/com/google/codeu/data/Datastore.java:47]: good catch! just fixed this

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 3 above)
Comment (donatKapesa) [reply src/main/java/com/google/codeu/data/Datastore.java:49]: yes they can. I was thinking on still saving them in the database as null and deal with this as I fetch messages in the front end. So I will have a something that checks for null imageUrls and conversationTopicIds. Does that work/make sense?

**Pull request closed without merging**